source/usr/lib/mcvirt/rpc/ssl_socket.py

```diff
@@ -26,21 +26,38 @@
 class SSLSocket(object):
     """Provides methods for wrapping Pyro methods with SSL"""
 
+    CIPHERS = ('ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:' +
+               'ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:' +
+               'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:' +
+               'DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:' +
+               'ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:' +
+               'ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:' +
+               'ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:' +
+               'DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:' +
+               'ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:' +
+               'AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS')
+
     @staticmethod
     def wrap_socket(socket_object, *args, **kwargs):
         """Wrap a Pyro socket connection with SSL"""
         server_side = ('bind' in kwargs.keys())
         ssl_kwargs = {
             'do_handshake_on_connect': True,
-            'ssl_version': ssl.PROTOCOL_TLSv1,
             'server_side': server_side
         }
+
+        # Create an SSL context
+        ssl_context = ssl.create_default_context()
+        ssl_context.set_ciphers(SSLSocket.CIPHERS)
+        ssl_context.verify_mode = ssl.CERT_OPTIONAL
```

Comment:
In the previous code it was only set to CERT_REQUIRED for client side sockets.

I changed it to CERT_REQUIRED but now getting this

`cannot connect: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:581)`

I think this is because for server side sockets we don't have the client's public key to verify against? My understanding of SSL is a bit limited...